Adding precision issue note docs for `functional.interpolate`

yewentao256 opened the pull request:
Fixes #104157

yewentao256 committed:
docs: adding note for the precision issue

lezcano commented:
There's a number of unrelated changes in the PR, perhaps an incorrect rebase?

yewentao256 commented:
> There's a number of unrelated changes in the PR, perhaps an incorrect rebase?

It seems so... Should I close this pr and fixes that issue, and make a pr again?

vfdev-5 commented:
> > There's a number of unrelated changes in the PR, perhaps an incorrect rebase?
> 
> It seems so... Should I close this pr and fixes that issue, and make a pr again?

no, you can just rebase correctly and push force to your fork

vfdev-5 commented on torch/nn/functional.py:
```diff
@@ -3878,6 +3878,12 @@ def interpolate(input: Tensor, size: Optional[int] = None, scale_factor: Optiona
+    .. note::
+        The use of ``float16`` data type in the backward pass of the upsample operation, specifically
+        for modes such as ``['linear', 'bilinear', 'bicubic', 'trilinear', 'area']``, could potentially
+        lead to precision issues. For instance, any result that exceeds 2048 will be truncated.
+        For more details, please refer to the discussion in `issue#104157 <https://github.com/pytorch/pytorch/issues/104157>`_.
```
I do not think we should mention 2048 here as it is specific to your repro example from the issue.

yewentao256 replied:
Done

yewentao256 commented:
> > > There's a number of unrelated changes in the PR, perhaps an incorrect rebase?
> > 
> > 
> > It seems so... Should I close this pr and fixes that issue, and make a pr again?
> 
> no, you can just rebase correctly and push force to your fork

It seems I've made another incorrect rebase 😂. I just rebased to the main branch of the PyTorch repository and pushed to my fork. I sincerely apologize for the mistake! Here's the command sequence I used:
```bash
git remote -v
myfork  git@github.com:yewentao256/pytorch.git (fetch)
myfork  git@github.com:yewentao256/pytorch.git (push)
origin  git@github.com:pytorch/pytorch.git (fetch)
origin  git@github.com:pytorch/pytorch.git (push)
(base) PS C:\Users\Peter\Desktop\pytorch> git rebase origin main
dropping 16d268e280514b61f8b731893b7bc6270c2e3b7b Fix comment error in TensorIterator.cpp (#100227) -- patch contents already upstream
Successfully rebased and updated refs/heads/main.
(base) PS C:\Users\Peter\Desktop\pytorch> git push myfork main --force
```
Is there anything I can do to fix this problem now?

vfdev-5 commented:
@yewentao256 try to do the following:
```
git reset --hard HEAD~2500
git pull origin main -r
git cherry-pick e015d03
# at this point you should have a linear history + one your commit with docs update
git log --oneline --decorate --graph

git push myfork main -f
```

A better practice is to create a branch and commit to a branch instead of `main`.

yewentao256 commented:
> @yewentao256 try to do the following:
> 
> ```
> git reset --hard HEAD~2500
> git pull origin main -r
> git cherry-pick e015d03
> # at this point you should have a linear history + one your commit with docs update
> git log --oneline --decorate --graph
> 
> git push myfork main -f
> ```
> 
> A better practice is to create a branch and commit to a branch instead of `main`.

@vfdev-5  Thanks a lot! Solved the issue👍

vfdev-5 commented:
Now, please address this comment: https://github.com/pytorch/pytorch/pull/104622#discussion_r1252991212

yewentao256 committed:
docs: remove 2048 example

yewentao256 commented:
> Now, please address this comment: [#104622 (comment)](https://github.com/pytorch/pytorch/pull/104622#discussion_r1252991212)

I removed the 2048 example there, is there anything else that I need to update? Please just let me know

vfdev-5 commented:
I would rephrase the message saying something like "gradients may be incorrect for dtype float16", I let your reword it in a better way. Also, I think this happens on CUDA only and not on CPU (can you please check?)

yewentao256 commented:
> I would rephrase the message saying something like "... gradients may be incorrect for dtype float16 ...", I let your reword it in a better way. IMO, the main message is "gradients can be incorrect". Also, I think this happens on CUDA only and not on CPU (can you please check?)

Yes, you are right, this issue will not happen on CPU (not supported for float16 currently).

Here's the revised content:

        The gradients may be incorrect for dtype ``float16`` on CUDA in the backward pass of the upsample when using modes ``['linear', 'bilinear', 'bicubic', 'trilinear', 'area']``. For more details, please refer to the discussion in `issue#104157 <https://github.com/pytorch/pytorch/issues/104157>`_.

If this makes sense, I will push a new commit then.

yewentao256 committed:
docs: revise the content

yewentao256 commented:
> Yes, looks better for me. I would just remove "during the backward pass" as we already talking about gradients

Done

ezyang commented:
@pytorchbot merge
